projects/igniteui-angular/src/lib/data-operations/pivot-sort-strategy.ts

```diff
@@ -47,13 +48,17 @@ export class DefaultPivotSortingStrategy extends DefaultSortingStrategy {
         const allDimensions = grid.allDimensions;
         const enabledDimensions = allDimensions.filter(x => x && x.enabled);
         this.dimension = PivotUtil.flatten(enabledDimensions).find(x => x.memberName === key);
+        this.pivotConfiguration = grid.pivotConfiguration;
         const reverse = (dir === SortingDirection.Desc ? -1 : 1);
         const cmpFunc = (obj1, obj2) => this.compareObjects(obj1, obj2, key, reverse, ignoreCase, this.getFieldValue, isDate, isTime);
         return this.arraySort(data, cmpFunc);
     }
 
     protected getFieldValue(obj: any, key: string, isDate: boolean = false, isTime: boolean = false): any {
-        let resolvedValue = PivotUtil.extractValueFromDimension(this.dimension, obj) || obj[0];
+        const isColumn = PivotUtil.flatten(this.pivotConfiguration.columns).find(dim => dim.memberName === this.dimension.memberName);
```

Comment:
Not a fan of conditionally changing the way the data is retrieved. obj should always be the original data object IMO.